[ARC] C compiler error when passing a var openArray to a sink openArray

This code results in a C compiler error. Found in https://github.com/bung87/tim_sort. I don't know if this code is supposed to work or not :) 

### Example
```nim
proc countRun(lst: sink openArray[int]): int =
  discard

proc timSort(lst: var openArray[int]) = 
  let run = countRun(lst)

var a = @[3, 5, 4, 2, 6, 1]
timSort(a)
echo a
```

### Current Output
```
/home/dian/.cache/nim/tim_sort_d/@mtim_sort.nim.c: In function ‘timSort__iUO1FYw3YnPPl6rsf8bqVg’:
/home/dian/.cache/nim/tim_sort_d/@mtim_sort.nim.c:156:54: error: ‘colontmpD_Len_0’ undeclared (first use in this function); did you mean ‘colontmpD_’?
  156 |  run = countRun__qbT9ayGxGQuz6mSoZvl6XYg(colontmpD_, colontmpD_Len_0);
      |                                                      ^~~~~~~~~~~~~~~
      |                                                      colontmpD_
```

### Expected Output
Not sure, it should either compile (if that is valid code), or result in a compiler error message (if we can't sink openArray)

```
$ nim -v
Nim Compiler Version 1.3.5 [Linux: amd64]
Compiled at 2020-07-21
Copyright (c) 2006-2020 by Andreas Rumpf

git hash: 911e39351e34c1022b586fe9bfc3a67b2ce3bf44
active boot switches: -d:release -d:danger
```

--- a/compiler/injectdestructors.nim
+++ b/compiler/injectdestructors.nim
@@ -699,9 +699,8 @@ proc p(n: PNode; c: var Con; s: var Scope; mode: ProcessMode): PNode =
     elif n.kind in {nkObjDownConv, nkObjUpConv}:
       result = copyTree(n)
       result[0] = p(n[0], c, s, sinkArg)
-    elif n.typ == nil or n.typ.skipTypes({tyGenericInst, tyAlias, tySink, tyVar}).kind == tyOpenArray:
-      # 'raise X' can be part of a 'case' expression. Deal with it here.
-      # openarrays require perfect forwarding
+    elif n.typ == nil:
+      # 'raise X' can be part of a 'case' expression. Deal with it here:
       result = p(n, c, s, normal)
     else:
       # copy objects that are not temporary but passed to a 'sink' parameter

--- a/tests/arc/t14383.nim
+++ b/tests/arc/t14383.nim
@@ -3,7 +3,6 @@ discard """
   output: '''
 hello
 hello
-@[4, 3, 2, 1]
 @["a", "b"]
 '''
 """
@@ -13,22 +12,6 @@ import dmodule
 var val = parseMinValue()
 if val.kind == minDictionary:
   echo val
-
-
-#------------------------------------------------------------------------------
-# Issue #15035
-#------------------------------------------------------------------------------
-
-proc countRun(lst: sink openArray[int]): int =
-  discard
-
-proc timSort(lst: var openArray[int]) = 
-  let run = countRun(lst)
-
-var a = @[4, 3, 2, 1]
-timSort(a)
-echo a
-
 
 #------------------------------------------------------------------------------
 # Issue #15238

